frontend/packages/kubevirt-plugin/integration-tests/tests/vm.cloud-init.sceario.ts

```diff
@@ -0,0 +1,61 @@
+import { execSync } from 'child_process';
+import { testName } from '../../../../integration-tests/protractor.conf';
+import { removeLeakedResources } from '../../../console-shared/src/test-utils/utils';
+import { statusIcons, waitForStatusIcon } from '../views/virtualMachine.view';
+import { VirtualMachine } from './models/virtualMachine';
+import { VM_BOOTUP_TIMEOUT_SECS } from './utils/consts';
+import { basicVMConfig } from './utils/mocks';
+import { CloudInitConfig, ProvisionConfig } from '../utils/types';
+import { vmConfig, CONFIG_NAME_CONTAINER } from './vm.wizard.configs';
+
+
+describe('Kubevirt create VM using cloud-init', () => {
+  const sourceContainer = 'kubevirt/fedora-cloud-container-disk-demo';
+  const provisionConfig: ProvisionConfig = {
+    provision: {
+      method: CONFIG_NAME_CONTAINER,
+      source: sourceContainer,
+    },
+    networkResources: [],
+    storageResources: [],
+  };
+
+  const hostnameAndKey: CloudInitConfig = {
+    useCloudInit: true,
+    hostname: 'fedora-kubevirt',
+    sshKey: 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCj47ubVnxR16JU7ZfDli3N5QVBAwJBRh2xMryyjk5dtfugo5JIPGB2cyXTqEDdzuRmI+Vkb/A5duJyBRlA+9RndGGmhhMnj8and3wu5/cEb7DkF6ZJ25QV4LQx3K/i57LStUHXRTvruHOZ2nCuVXWqi7wSvz5YcvEv7O8pNF5uGmqHlShBdxQxcjurXACZ1YY0YDJDr3AJai1KF9zehVJODuSbrnOYpThVWGjFuFAnNxbtuZ8EOSougN2aYTf2qr/KFGDHtewIkzZmP6cjzKO5bN3pVbXxmb2Gces/BYHntY4MXBTUqwsmsCRC5SAz14bEP/vsLtrNhjq9vCS+BjMT',
+  };
+  const customScript: CloudInitConfig = {
+    useCloudInit: true,
+    useCustomScript: true,
+    customScript: basicVMConfig.cloudInitScript,
+  };
+
+  const getVolumes = (vmName: string) => {
+    return `kubectl get -n ${testName} vm ${vmName} -o jsonpath='{.spec.template.spec.volumes}'`;
+  };
+
+  it('Create vm using hostname and key as cloud-init', async () => {
+    const vmName = 'vm-cloudinit-hostname-key';
+    const testVMConfig = vmConfig(vmName, provisionConfig, testName);
+    testVMConfig.cloudInit = hostnameAndKey;
+
+    const vm = new VirtualMachine(testVMConfig);
+    await vm.create(testVMConfig);
+
+    const volumes = execSync(getVolumes(vm.name)).toString();
```

Comment:
Could you use `getVolumes(getResourceObject(vm.name, vm.namespace, vm.kind))`
with `getVolumes` method taken from `packages/kubevirt-plugin/src/selectors/vm/selectors.ts`.
Then it's not necessary to reimplement `getVolumes` on line 34

Reply:
updated by using a similar method in selectors.ts.